app/validators/sections/section_validator.py

```diff
@@ -177,14 +153,7 @@ def validate_variants(self, block):
 
         for variant in all_variants:
             when_clause = variant.get("when", [])
-
-            if isinstance(when_clause, list):
-                when_validator = WhenRuleValidator(
-                    when_clause, block["id"], self.questionnaire_schema
-                )
-                self.errors += when_validator.validate()
-
-            elif isinstance(when_clause, dict):
+            if isinstance(when_clause, dict):
```

Comment:
Same as above. Should be a dict?. The above lines default for `.get()` needs to change.

Reply:
Replaced with 

```python
if when_clause := variant.get("when", None):
```